cpp/thirdparty/versions.txt

```diff
@@ -25,35 +25,39 @@
 
 # Abseil LTS branch, Sept 2020, Patch 2
 ARROW_ABSL_BUILD_VERSION=0f3bb466b868b523cf1dc9b2aaaed65c77b28862
-ARROW_AWSSDK_BUILD_VERSION=1.8.90
+ARROW_AWSSDK_BUILD_VERSION=1.8.128
+# aws-checksums 0.1.10 has some issues it seems
 ARROW_AWS_CHECKSUMS_BUILD_VERSION=v0.1.5
 ARROW_AWS_C_COMMON_BUILD_VERSION=v0.4.59
 ARROW_AWS_C_EVENT_STREAM_BUILD_VERSION=v0.1.5
 ARROW_BOOST_BUILD_VERSION=1.71.0
-ARROW_BROTLI_BUILD_VERSION=v1.0.7
+ARROW_BROTLI_BUILD_VERSION=v1.0.9
 ARROW_BZIP2_BUILD_VERSION=1.0.8
-ARROW_CARES_BUILD_VERSION=1.16.1
+ARROW_CARES_BUILD_VERSION=1.17.1
 ARROW_GBENCHMARK_BUILD_VERSION=v1.5.2
 ARROW_GFLAGS_BUILD_VERSION=v2.2.2
 ARROW_GLOG_BUILD_VERSION=v0.4.0
 ARROW_GRPC_BUILD_VERSION=v1.33.2
 ARROW_GTEST_BUILD_VERSION=1.10.0
 ARROW_JEMALLOC_BUILD_VERSION=5.2.1
 ARROW_LZ4_BUILD_VERSION=v1.9.2
+# mimalloc 1.6.7 didn't build on Visual Studio 2015
```

Comment:
Did you report the issue upstream?

Reply:
Not yet; error was `C:\projects\arrow\cpp\build\mimalloc_ep-prefix\src\mimalloc_ep\src\heap.c(357): error C2593: 'operator ==' is ambiguous`, what does that suggest to you? 

https://ci.appveyor.com/project/nealrichardson/arrow/builds/37402650/job/4uag688i5yflo1c3#L981

Reply:
This suggest this should be fixed upstream :-)

Reply:
https://github.com/microsoft/mimalloc/issues/353

Reply:
There's a fix proposed here: https://github.com/microsoft/mimalloc/issues/353#issuecomment-770085626

Someone with better cmake skills than I could try to pull https://github.com/microsoft/mimalloc/commit/f68c1a74da7279f32550fe0b1eeb334079bf2687.patch in as patch step

Reply:
We've worked out a patch over on the mimalloc PR, but I'll deal with that in a followup if it's ok. Looks like there's a 2.0 release coming, and some other Apple M1 fixes coming, so we should probably just upgrade when those are in.